subhub/subhub_dynamodb.py

```diff
@@ -69,5 +69,64 @@ def remove_from_db(self, uid: str) -> bool:
         try:
             self.model.get(uid, consistent_read=True).delete()
             return True
-        except DoesNotExist as e:
+        except DoesNotExist:
+            return False
+
+
+class WebHookEventModel(Model):
+    eventId = UnicodeAttribute(hash_key=True)
+    sent_system = ListAttribute(default=list)
+
+
+class WebHookEvent:
+    def __init__(self, table_name: str, region: str, host: Optional[str] = None):
+        _table = table_name
+        _region = region
+        _host = host
+
+        class WebHookEventModel(Model):
+            class Meta:
+                table_name = _table
+                region = _region
+                if _host:
+                    host = _host
+
+            eventId = UnicodeAttribute(hash_key=True)
+            sent_system = ListAttribute()
+
+        self.model = WebHookEventModel
+
+    def new_event(self, event_id: str, sent_system: list) -> WebHookEventModel:
+        return self.model(eventId=event_id, sent_system=[sent_system])
+
+    def get_event(self, event_id: str) -> Optional[WebHookEventModel]:
+        try:
+            webhook_event = self.model.get(event_id, consistent_read=True)
+            return webhook_event
+        except DoesNotExist:
+            return None
+
+    @staticmethod
+    def save_event(webhook_event: WebHookEventModel) -> bool:
+        try:
+            webhook_event.save()
+            return True
+        except PutError:
+            return False
+
+    def append_event(self, event_id: str, sent_system: str) -> bool:
+        try:
+            update_event = self.model.get(event_id, consistent_read=True)
+            if not sent_system in update_event.sent_system:
+                update_event.sent_system.append(sent_system)
+                update_event.save()
+                return True
+        except DoesNotExist:
+            return False
+
+    def remove_from_db(self, event_id: str) -> bool:
+        try:
+            self.model.get(event_id, consistent_read=True).delete()
```

Comment:
It should be possible to delete without fetching first which will reduce cost and round trips.